plugins/registry.js

```diff
@@ -329,6 +329,43 @@ export default class PluginRegistry {
         return id;
     }
 
+    // Register a hook that will be called when a slash command is posted by the user before it
+    // is sent to the server. Accepts a function that receives the message (string) and the args
+    // (object) as arguments.
+    // The args object is:
+    //        {
+    //            channel_id: channelId,
+    //            team_id: teamId,
+    //            root_id: rootId,
+    //            parent_id: rootId,
+    //        }
+    //
+    // To reject a command, return an object containing an error:
+    //     {error: {message: 'Rejected'}}
+    // To ignore a command, return an empty object (to prevent an error from being displayed):
+    //     {}
+    // To modify or allow the command without modification, return an object containing the new message
+    // and args. It is not likely that you will need to change the args, so return the object that was provided:
+    //     {message: {...}, args}
+    //
+    // If the hook function is asynchronous, the command will not be sent to the server
```

Comment:
@cpoile Should we note the minimum webapp version this change applies to? (5.12+)

Reply:
@jasonblais Sure, how should we do that? Right here in the doc string?

Reply:
@cpoile Including right here in the doc works, we do so for plugin API minimum version notices https://github.com/mattermost/mattermost-server/blob/master/plugin/api.go#L133

Seems we haven't done this for other client-side hooks, as you noted, but it would be good to start doing it.